File: 2026-1/sesion-de-clase-03/java/seguro-vehicular/CoberturaContraRobos.java

public class CoberturaContraRobos extends SeguroDecorador {

    public CoberturaContraRobos(Cotizable seguro) {
        super(seguro);
    }

    @Override
    public double CalcularCosto() {
        double costo = 100.00;

        if (this.seguro.Vehiculo().getMarca() == Marca.Toyota) {
            costo += 100.00;
        }

        return this.seguro.CalcularCosto() + costo;
    }

    @Override
    public String Descripcion() {
        return this.seguro.Descripcion() + ", cobertura contra robos";
    }
}

File: 2026-1/sesion-de-clase-05/dotnet/seguroslib/Coberturas/CoberturaContraRobos.cs
namespace Coberturas;

using Modelo;

public class CoberturaContraRobos : SeguroDecorador
{
    public CoberturaContraRobos(ICotizable seguro) : base(seguro)
    {
    }

    public override double CalcularCosto()
    {
        double costo = 100.00;

        if (this.seguro.Vehiculo().Marca == Marca.Toyota)
        {
            costo += 100.00;
        }

        return this.seguro.CalcularCosto() + costo;
    }

    public override string Descripcion()
    {
        return this.seguro.Descripcion() + ", cobertura contra robos";
    }
}
